### PR TITLE
Pull request for #85: Find way to speed up log.py

### PR DESCRIPTION
**Pull request by arosenbe**
@stanfordquan can you please PR my work from #85? I replaced `os.walk` with a (platform-specific) subprocess calling a command line tool used to search for files with a certain name. Let me know if you have any questions!

### CODE REVIEW

File: gslab_scons/log.py [hunk 1]
```diff
@@ -69,27 +70,38 @@ def log_timestamp(start_time, end_time, filename = 'sconstruct.log'):
...
+    # Store paths to logs in a list, found from platform-specific command line tool 
+    rel_parent_dir = os.path.relpath(parent_dir)
+    log_name = 'sconscript*.log'
+    if misc.is_unix():
+        cmd = 'find %s -name "%s"' % (rel_parent_dir, log_name)
+    else:
+        cmd = 'dir "%s" /b/s' % os.path.join(rel_parent_dir, log_name)
+    try:
+        log_paths = subprocess.check_output(cmd, shell = True).replace('\r\n', '\n')
+        log_paths = log_paths.split('\n')[:-1] # Last entry always ''
```
Comment (stanfordquan) [inline gslab_scons/log.py]: please assert this before subsetting (perhaps with an if statement)

File: gslab_scons/log.py [hunk 2]
```diff
@@ -52,7 +53,7 @@ def end_log(log = 'sconstruct.log'):
             with open(f, 'rU') as sconscript:
                 if this_run_dict[f] == beginning_of_time:
                     sconstruct.write("*** Warning!!! Doesn't look the sconscript below finished.\n")
```
Comment (stanfordquan) [inline gslab_scons/log.py]: Can you please do me a favor and change this one to " Warning!!! The log below does not have timestamps, the Sconscript may not have finished."

File: gslab_scons/log.py [hunk 3]
```diff
@@ -69,27 +70,38 @@ def log_timestamp(start_time, end_time, filename = 'sconstruct.log'):
...
-                        builder_log_end_time = datetime.strptime(s, "%Y-%m-%d %H:%M:%S")
-                    except ValueError: # if the code breaks, there's no time identifier
-                        beginning_of_time    = datetime.min
-                        builder_log_end_time = beginning_of_time
-                builder_log_collect[f_full]  = builder_log_end_time
+    # Store paths to logs in a list, found from platform-specific command line tool 
+    rel_parent_dir = os.path.relpath(parent_dir)
+    log_name = 'sconscript*.log'
+    if misc.is_unix():
+        cmd = 'find %s -name "%s"' % (rel_parent_dir, log_name)
```
Comment (arosenbe) [inline gslab_scons/log.py]: Now called `command`, thanks for catching that silly name.

Comment (arosenbe) [inline gslab_scons/log.py]: It does catch that name on Mac. 

```bash
mkdir temp # Watch paths
mkdir temp/temp2
touch "temp/temp2/sconscript_@\!\#ASD.log"
find . -name "sconscript*.log"
rm -rf temp # BEWARE
```

File: gslab_scons/log.py (hunk 1 above)
Comment (arosenbe) [inline gslab_scons/log.py]: Uses filter and map to avoid hard coding anything.

File: gslab_scons/log.py (hunk 2 above)
Comment (arosenbe) [inline gslab_scons/log.py]: Sure thing. I assumed you wanted to keep the new line at the end, but let me know if not.

**Comment by arosenbe** [comment]
Thanks for the good suggestions @stanfordquan! I think https://github.com/gslab-econ/gslab_python/pull/87/commits/0c67612960deb39cd7788b6f5239ef05d8a951d3 above takes care of them, but let me know if not.

**Pull request merged**